gprofiler/profilers/factory.py

```diff
@@ -0,0 +1,38 @@
+from typing import TYPE_CHECKING, Any, List, Tuple, Union
+
+from gprofiler.log import get_logger_adapter
+from gprofiler.profilers.profiler_base import NoopProfiler
+
+if TYPE_CHECKING:
+    from gprofiler.main import UserArgs
+    from gprofiler.profilers.profiler_base import ProcessProfilerBase
+
+from gprofiler.profilers.perf import SystemProfiler
+from gprofiler.profilers.registry import get_profilers_registry
+
+logger = get_logger_adapter(__name__)
+
+
+def get_profilers(
+    user_args: 'UserArgs', **profiler_init_kwargs: Any
+) -> Tuple[Union['SystemProfiler', 'NoopProfiler'], List['ProcessProfilerBase']]:
```

Comment:
```suggestion
) -> Tuple[Union[SystemProfiler, NoopProfiler], List[ProcessProfilerBase]]:
```
might need to add `from __future__ import annotations`?

Reply:
This future doesn't exist in Python 3.6 I believe :/

Reply:
Blehhh